### PR TITLE
Basic Chaos Rework, Confusion generic effect

### PR DESCRIPTION
**Pull request by Shushishtok**
- Added the Confusion generic effect, which forces units afflicted with it to move to random positions around them while disabling all other orders.
    - Player controlled units are command restricted while they wander.
    - Enemy units simply override their logic and are forced to use the WanderAround() logic.
- Reworked Basic Chaos. Enemies that are exposed to the laser for a long enough time during a single shot now become afflicted with Confusion for a moderate duration. Multiple Dark Lasers coming in contacts with the same targets can speed the process up.
- Added a few debugging commands and constants to assist with debugging.
- Moved the wander around weights to constants to be accessed from both the Confusion and the Wave Modifiers.
- Fixed a bug where `Arena.FindPositionInside` always resulted in a distance of 100 (was hardcoded).
- Reworked the move speed bonus to be a percentage increase without relying on recording the base speed of the unit. It now should always reach 100% bonus move speed in 20 seconds.
- Reworked how Overtime is applied in the wave modifier logic. Instead of changing the AI logic to touch only when they try to run away, enemies now try touching the player instead. 
    - This had a side effect where units with abilities stopped casting them completely. This essentially changes units to have the following logic during Overtime:
        - Run away -> Touch Only
        - Cast only -> Cast and Touch
- Hidden the laser emitter modifier.

### CODE REVIEW

File: game/scripts/vscripts/core/generic_weapon_effects.ts [hunk 1]
```diff
@@ -190,6 +202,47 @@ export namespace GenericWeaponEffect {
...
+	 * @param unit The unit to apply the modifier to.
+	 * @param duration The duration of the confusion modifier to be applied.
+	 */
+	export function ApplyConfusionWeaponEffect(weapon: BaseWeapon, unit: CDOTA_BaseNPC, duration: number) {
+		// If the modifier exists and the current duration is lower than the new attempted duration, set new duration and refresh.
+		const modifier = unit.FindModifierByName(modifier_chaos_confusion.name);
+		if (modifier) {
+			const currentDuration = modifier.GetRemainingTime();
+			if (duration > currentDuration) {
+				modifier.SetDuration(duration, true);
```
Comment (Nibuja05) [inline game/scripts/vscripts/core/generic_weapon_effects.ts:221]: Maybe half the duration for player characters?

File: game/scripts/vscripts/core/generic_weapon_effects.ts [hunk 2]
```diff
@@ -485,3 +538,74 @@ class modifier_electricity_zapped extends WeaponModifier {
...
+		return true;
+	}
+
+	IsPurgable() {
+		return true;
+	}
+
+	// This is only called if the unit is player controlled.
+	OnIntervalThink() {
+		if (this.parent.IsMoving()) return;
```
Comment (Nibuja05) [inline game/scripts/vscripts/core/generic_weapon_effects.ts:577]: Makes the movement unfluent, please remove

File: game/scripts/vscripts/core/generic_weapon_effects.ts [hunk 3]
```diff
@@ -485,3 +538,74 @@ class modifier_electricity_zapped extends WeaponModifier {
...
+		const centerDirection = Util.CalculateDirection(parentLoc, center);
+		let direction = (randomDirection * weights.random + centerDirection * weights.center) as Vector;
+		if (this.previousDirection) {
+			direction = (direction + this.previousDirection * weights.previous) as Vector;
+		}
+		direction = direction.Normalized();
+		let [runToPoint, finalDirection] = this.arena.FindPositionInside(
+			parentLoc,
+			direction,
+			RandomInt(this.minDistance, this.maxDistance)
```
Comment (Nibuja05) [inline game/scripts/vscripts/core/generic_weapon_effects.ts:601]: Please always use 300 here (for higher movespeeds)

**Review by Nibuja05 — COMMENTED** [review]
Only things we already discussed need to be changed :)

File: game/scripts/vscripts/core/generic_weapon_effects.ts (hunk 1 above)
Comment (Shushishtok) [reply game/scripts/vscripts/core/generic_weapon_effects.ts:221]: Done.

File: game/scripts/vscripts/core/generic_weapon_effects.ts (hunk 2 above)
Comment (Shushishtok) [reply game/scripts/vscripts/core/generic_weapon_effects.ts:577]: Done.

File: game/scripts/vscripts/core/generic_weapon_effects.ts (hunk 3 above)
Comment (Shushishtok) [reply game/scripts/vscripts/core/generic_weapon_effects.ts:601]: Done.

**Review by Nibuja05 — APPROVED** [review]

**Pull request merged**